### PR TITLE
Adding python bindings for ecl io class ERst

### PR DESCRIPTION
**Pull request by tskille**

### CODE REVIEW

File: python/cxx/eclipse_io.cpp [hunk 1]
```diff
@@ -33,6 +37,105 @@ py::array get_vector(Opm::EclIO::EclFile * file_ptr, std::size_t array_index) {
     throw std::logic_error("Data type not supported");
 }
 
+bool erst_contains(Opm::EclIO::ERst * file_ptr, std::tuple<std::string, int> keyword) {
+
+    bool hasKeyAtReport = file_ptr->count(std::get<0>(keyword), std::get<1>(keyword)) > 0 ? true : false;
+
```
Comment (joakim-hove) [inline python/cxx/eclipse_io.cpp]: The use of blank lines should help the reader understand the structure in the code - your blank seems to be spread out a bit randomly?

File: python/tests/test_erst.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        info_str = "List of report steps\n\nseqnum =   37 Date: 06-Jan-2016\nseqnum =   74 Date: 10-Jan-2017\n"
+
+        rst1 = ERst(test_path("data/SPE9.UNRST"))
+
+        self.assertTrue(repr(rst1) == info_str)
+
+        self.assertTrue(rst1.has_report_step_number(37))
+        self.assertTrue(rst1.has_report_step_number(74))
+
+        self.assertRaises(ValueError, rst1.load_report_step, 137 );
```
Comment (joakim-hove) [inline python/tests/test_erst.py]: The most common way to express these tests is:
```python
with self.assertRaises(ValueError):
      rst1.load_report_step( 137 )
```

Comment (tskille) [reply python/tests/test_erst.py]: @joakim-hove can you please let me know if this will be merged?  When do you expect to finish reviewing this PR ?

**Comment by joakim-hove** [comment]
> @joakim-hove can you please let me know if this will be merged? When do you expect to finish reviewing this PR?

I am certainly positive to the Python wrappers - obviously - and eventually intend to merge this. However I am not entirely happy with this PR in it's current form - and have shied a bit away from reviewing it. My main criticism is not with this PR as such - but rather the underlying C++ classes, and specifically their internal relationship.

As I see it the lowest level class `EclFile` does not have good enough internal indexing primitives, and the implementation details of `EclFile` leak too much up to the`ERst` class; furthermore much of the "get a keyword" functionality is reimplemented in the `ERst` - it seems the `EclFile` class is only partly used as a base class? These problems stand out more when the classes are wrapped in Python.

**Pull request closed without merging**